src/snapshot_middleware/json_model.rs

```diff
@@ -18,6 +18,17 @@ pub fn snapshot_json_model(
     instance_name: &str,
 ) -> SnapshotInstanceResult {
     let contents = vfs.read(path)?;
+    if contents.is_empty() {
+        return Ok(None);
+    }
+
+    if match String::from_utf8((*contents).to_owned()) {
```

Comment:
Why not str::from_utf8, rather than allocating an entirely new string here?

Reply:
Didn't know about/look up str::from_utf8, I've changed it to use str::from_utf8 now